application-mocca-calendar-ui/src/main/resources/MoccaCalendar/Code/Macro.xml

```diff
@@ -896,6 +910,73 @@ require(['jquery'], function($) {
     var calendarPopup = relatedTarget.data('calendarPopup');
     new XWiki.MoccaCalendar.MoccaCalendarPopup(calendarPopup.interactionParameters, calendarPopup.helper);
   });
+
+  // Trigger the upload manually when the button is clicked
+  $(document).on('click', '#import-calendar-file-button', function(event) {
+    var form = $('#import-calendar-file form').get(0);
+    if (event) {
+      event.preventDefault();
+    }
+    startUploading(form);
+  });
+
+  // Start uploading this file, creating a new XHR object with the file data.
+  var startUploading = function (form) {
+    var formData = new FormData(form);
+    var select = form.down("#import-calendar-parent");
+    var input = form.down("#import-ical-file-input");
+    if (input.files[0].size &lt; input.dataset.maxFileSize) {
+      var urlQueryParameter = "?" + select.name + "=" + select.value;
+      formData.action = form.action + urlQueryParameter;
```

Comment:
I would use [$.param](https://api.jquery.com/jQuery.param/) for creating the query params.